lib/codegen/src/isa/x86/abi.rs

```diff
@@ -425,6 +425,21 @@ fn insert_common_prologue(
     csrs: &RegisterSet,
     isa: &TargetIsa,
 ) {
+    if stack_size > 0 {
+        // Check if there is a special stack limit parameter. If so insert stack check.
+        if let Some(stack_limit_arg) = pos.func.special_param(ArgumentPurpose::StackLimit) {
+            // Total stack size is the size of all stack area used by the function, including
+            // pushed CSRs, frame pointer. 
+            // Also, the size of a return address, implicitly pushed by a x86 `call` instruction, also 
+            // should be accounted for.
+            // TODO: Check if the function body actually contains a `call` instruction.
+            let word_size = isa.pointer_bytes();
+            let total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64;
```

Comment:
For now, it pessimistically accounts for an implicit return address pushed by a call instruction.

What is the best way to find out if the function contains a call instruction? Iterate over all instructions here?

Reply:
Why is `+ 2` written as `+ 1 + 1`?

Reply:
It is just simply because I wanted to make it clear that one is for frame pointer register and another one for a return address. But apparently I've failed : D

Reply:
The absolute simplest way would be to just assume there's always a call.

Beyond that, iterating through the instructions is the best we can do for now.